IMAP: Gmail: Invalid credentials (Failure) in some sessions

1. Go to Mail
2. Click to an account
3. Click to Sent Mail

Actual result
Instead of the content there is 'Invalid credential (Failure)'

Expected result
Seeing the content of the sent mail



![image](https://github.com/user-attachments/assets/3d614938-0d4f-4388-a20d-4c4a0770d000)

Leaving the app open idling also produces the following errors:

```
Error: <email address>: Connection not available
    at message.js:155:16
    at new Promise (<anonymous>)
    at WSCall.makeCall (message.js:150:12)
    at JPCWebSocket.callRemote (protocol.js:184:31)
    at RemoteClass.next (obj.js:240:55)
    at IMAPFolder.ts:322:24
    at async IMAPFolder.runCommand (IMAPFolder.ts:85:14)
    at async IMAPFolder.downloadMessages (IMAPFolder.ts:312:13)
    at async IMAPFolder.getNewMessages (IMAPFolder.ts:288:11)
    at async emailAccountLogin (startup.ts:68:11)
```

```
Opening IMAP folder failed Error: Connection not available
    at message.js:155:16
    at new Promise (<anonymous>)
    at WSCall.makeCall (message.js:150:12)
    at JPCWebSocket.callRemote (protocol.js:184:31)
    at RemoteClass.getMailboxLock (obj.js:240:55)
    at IMAPFolder.runCommand (IMAPFolder.ts:65:29)
    at async IMAPFolder.fetchUIDList (IMAPFolder.ts:267:11)
    at async IMAPFolder.checkDeletedMessages (IMAPFolder.ts:451:22)
    at async IMAPFolder.getNewMessages (IMAPFolder.ts:286:5)
    at async IMAPFolder.pollRun (IMAPFolder.ts:438:5)
    at async IMAPFolder.ts:422:9
```

```
error.ts:46 Error: Connection purpose unknown
    at assert$2 (util.ts:4:11)
    at IMAPAccount.reconnect (IMAPAccount.ts:226:5)
    at IMAPFolder.runCommand (IMAPFolder.ts:74:37)
    at async IMAPFolder.fetchUIDList (IMAPFolder.ts:267:11)
    at async IMAPFolder.checkDeletedMessages (IMAPFolder.ts:451:22)
    at async IMAPFolder.getNewMessages (IMAPFolder.ts:286:5)
    at async IMAPFolder.pollRun (IMAPFolder.ts:438:5)
    at async IMAPFolder.ts:422:9
```

--- a/app/logic/Auth/OAuth2.ts
+++ b/app/logic/Auth/OAuth2.ts
@@ -92,7 +92,7 @@ export class OAuth2 extends WebBasedAuth {
    */
   async login(interactive: boolean): Promise<string> {
     assert(this.account, "Need to set account first");
-    if (this.accessToken) {
+    if (this.isLoggedIn) {
       return this.accessToken;
     }
     this.refreshToken ??= await this.getRefreshTokenFromStorage();

